File: python/16c_344A_Magnets.py
def main():
    prev, next, grp = 0, 0, 0
    n = int(input())
    for i in range(n):
        temp = input()
        if i == 0:
            prev = temp[0]
            continue
        next = temp[0]
        if next != prev:
            grp += 1
            prev = next
    grp += 1
    print(grp)

if __name__ == "__main__":
    main()

File: cpp/16c_344A_Magnets.cpp
#include<bits/stdc++.h>

int get_pole(std::string s){
    if ( s[0] == '0'){
        return 0;
    } 
    return 1;
}

int main(){
    int n, prev, next, grp = 0;
    std::string temp;
    std::cin>>n;
    for(int i = 0; i < n; i++){
        std::cin>>temp;
        if (i == 0){
            prev = get_pole(temp);
            continue;
        }
        next = get_pole(temp);
        if (next == prev){
            prev = next;
        }
        else{
            grp++;
            prev = next;
        }
    }
    grp++;
    std::cout<<grp;
    return 0;
}